symbolic-base/src/ZkFold/Symbolic/Data/UInt.hs

```diff
@@ -412,6 +413,18 @@ instance ( Symbolic c, KnownNat n, KnownRegisterSize r
           constraint (($ l) - one)
           return dm
 
+asWordsReg
+    :: forall regSize ctx k
+    .  Symbolic ctx
+    => KnownNat regSize
+    => ctx (Vector k)                           -- @k@ registers of size up to @regSize@
+    -> ctx (Vector k :.: Vector regSize) -- @k * wordsPerReg@ registers of size @wordSize@
+asWordsReg v = fromCircuitF v $ \regs -> do
+    words <- for regs $ \reg -> do
+      bits <- expansionW @regSize (value @regSize) reg
+      return (V.reverse (V.unsafeToVector bits))
+    return (Comp1 words)
+
```

Comment:
Not sure if this is the right way to expand. Is there a better way? I'm concerned about it incurring proliferating `KnownNat` constraints which is nothing new but annoying and feels like it shouldn't be necessary.